File: spring-data-requery-java/src/main/java/com/coupang/springframework/data/requery/repository/query/CommandType.java
package com.coupang.springframework.data.requery.repository.query;

import lombok.extern.slf4j.Slf4j;
import org.springframework.util.StringUtils;

/**
 * CommandType
 *
 * @author debop@coupang.com
 * @since 18. 6. 27
 */
@Slf4j
public enum CommandType {
    NONE,
    SELECT,
    INSERT,
    UPDATE,
    UPSERT,
    DELETE,
    REFRESH;


    public static CommandType parse(String methodName) {
        log.debug("Parse methodName={}", methodName);

        if (!StringUtils.hasText(methodName)) {
            return NONE;
        }
        String method = methodName.toLowerCase();

        if (method.startsWith("insert")) {
            return INSERT;
        }
        if (method.startsWith("update")) {
            return UPDATE;
        }
        if (method.startsWith("upsert")) {
            return UPSERT;
        }
        if (method.startsWith("delete") || method.startsWith("remove")) {
            return DELETE;
        }
        return SELECT;
    }
}

File: spring-data-requery-kotlin/src/main/kotlin/org/springframework/data/requery/kotlin/repository/query/CommandType.kt
package org.springframework.data.requery.repository.query

import mu.KotlinLogging

/**
 * org.springframework.data.requery.repository.query.CommandType
 *
 * @author debop
 */
enum class CommandType {

    NONE,
    SELECT,
    INSERT,
    UPDATE,
    UPSERT,
    DELETE,
    REFRESH;

    companion object {
        private val log = KotlinLogging.logger { }

        fun parse(command: String?): CommandType {
            if(command.isNullOrBlank())
                return NONE

            if(command!! == "remove") {
                return DELETE
            }
            return CommandType.values()
                       .find { it.name == command.toUpperCase() }
                   ?: SELECT
        }
    }
}